packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -887,24 +887,40 @@ class _FlutterPlatform extends PlatformPlugin {
 class _FlutterPlatformStreamSinkWrapper<S> implements StreamSink<S> {
   _FlutterPlatformStreamSinkWrapper(this._parent, this._shellProcessClosed);
   final StreamSink<S> _parent;
-  final Future<Null> _shellProcessClosed;
-
+  final Future<dynamic> _shellProcessClosed;
+
+  /// This object is created within [_FlutterPlatform.loadChannel], which is
+  /// called from a guard zone set up by the the test framework.  However,
+  /// [close] is called by the framework during the suite shutdown process,
+  /// which runs in a different zone.
+  ///
+  /// If the [done] future is allocated in the first zone, then Dart will not
+  /// allow [close] to complete that future with an error during a call from
+  /// the second zone.  To work around this, [close] allocates its own future, and
+  /// [done] is not supported.
```

Comment:
This doesn't seem like a clean solution. I'd rather we refactored this code to just make sure the exception gets thrown properly, e.g. by having the _done completer be created in the right zone and passed in, or some such.

Reply:
maybe we should just pass in the zone from the parent so that we can create things in that zone?